src/features/settings/components/BookmarkSettings/BookmarkSettings.tsx

```diff
@@ -60,17 +61,72 @@ const BookmarkItem = ({ item, appendRef = false }: BookmarkItemProps) => {
 }
 
 export const BookmarkSettings = () => {
-  const { userBookmarks } = useBookmarks()
+  const inputFile = useRef<HTMLInputElement | null>(null)
+  const { bookmarkPost, userBookmarks } = useBookmarks()
+  const [downloadURL, setDownloadURL] = useState('')
+  const downloadFileName = 'hackertabBookmarks'
+
+  useEffect(() => {
+    // Generating URL for exporting bookmarks
+    const bookmarkData = new Blob([JSON.stringify(userBookmarks, null, 2)], {
+      type: 'application/json',
+    })
+    setDownloadURL(URL.createObjectURL(bookmarkData))
+  }, [userBookmarks])
+
+  const ImportBookmarks = () => {
+    inputFile.current?.click()
+  }
+
+  const handleFileChange = (event: any) => {
+    if (event.target.files[0]) {
+      const reader = new FileReader()
+      reader.onload = () => {
+        const importData = JSON.parse(reader.result as string)
+        const uniqueImportData = importData.filter(
+          (data: BookmarkedPost) =>
+            !userBookmarks.some((bm) => bm.title === data.title && bm.url === data.url)
+        )
+        uniqueImportData.forEach((data: BookmarkedPost) => {
+          bookmarkPost(data)
+        })
+      }
+      reader.readAsText(event.target.files[0])
+      // HACK:sometimes the bookmark is getting updated on live so need to refresh the page
+      window.location.reload()
+    }
+  }
 
   return (
-    <SettingsContentLayout
-      title="Bookmarks"
-      description="Find all your bookmarks here. You can remove a bookmark by clicking on the remove icon.">
-      <div className="bookmarks">
-        {userBookmarks.map((bm) => (
-          <BookmarkItem item={bm} key={bm.url} />
-        ))}
+    <>
+      <div className="btn-group">
+        <div>
+          <input
+            type="file"
+            id="file"
+            ref={inputFile}
+            style={{ display: 'none' }}
+            onChange={handleFileChange}
+          />
+          <button id="OpenImgUpload" className="notbtn btn" onClick={ImportBookmarks}>
+            Import
+          </button>
+        </div>
+        <button className="notbtn btn">
+          <a href={downloadURL} download={downloadFileName} style={{ textDecoration: 'none' }}>
+            Export
+          </a>
+        </button>
```

Comment:
You can handle the click by using `onClick` of the button like you did for `importBookmarks` and remove the `a` to avoid confusion as both `button` and `a` have the `onClick` method.
Once done, you won't need the `useEffect` in the top.